analyses/mwi/dryspells_trigger/16_review_jan_feb_pred_threshold.md

```diff
@@ -0,0 +1,494 @@
+    df_ds_daterange["dryspell_obs"]=1
+    df_ds_daterange["date_month"]=df_ds_daterange.date.dt.to_period("M")
+    
+    #count the number of days within a year-month combination that had were part of a dry spell
+    df_ds_countmonth=df_ds_daterange.groupby([ds_adm_col,"date_month"],as_index=False).sum()
+    
+    df_ds_month=df_ds_countmonth[df_ds_countmonth.dryspell_obs>=min_ds_days_month]
+    
+    #TODO: this is not really generalizable
+    if shp_adm_col not in df_ds_month.columns:
+        df_adm2=gpd.read_file(adm2_bound_path)
+        df_ds_month=df_ds_month.merge(df_adm2[["ADM2_PCODE","ADM2_EN","ADM1_EN"]],left_on=ds_adm_col,right_on="ADM2_PCODE")
+        
+    df_ds_month_adm1=df_ds_month.groupby([shp_adm_col,"date_month"],as_index=False).count()
+    df_ds_month_adm1["dry_spell"]=np.where(df_ds_month_adm1.dryspell_obs>=min_adm_ds_month,1,0)
+    
+    return df_ds_month_adm1
+```
+
+### Load the forecast data
+
+
+```python
+start_year=2000
+end_year=2020
+#just locking the date to keep the analysis the same even though data is added
+#might wanna delete again later
+end_date="2-1-2020"
+```
+
+
+```python
+#read the ecmwf forecast per adm1 per date and concat all dates
+# the mwi_seasonal-monthly-single-levels_v5_interp*.csv contain results when interpolating the forecasts to be more granular
+# but results actually worsen with this
+date_list=pd.date_range(start=f'1-1-{start_year}', end=end_date, freq='MS')
+all_files=[processing.get_stats_filepath(country_iso3,config,date,resolution=resolution,
+                                         adm_level=1,use_unrounded_area_coords=use_unrounded_area_coords,
+                                         source_cds=source_cds,all_touched=all_touched) for date in date_list]
+
+df_from_each_file = (pd.read_csv(f,parse_dates=["date"]) for f in all_files)
+df_for   = pd.concat(df_from_each_file, ignore_index=True)
+
+```
+
+
+```python
+#for now using mean cell as this requires one variable less to be set (else need to set percentage of cells)
+#for earlier dates, the model included less members --> values for those members are nan --> remove those rows
+df_for = df_for[df_for[aggr_meth].notna()]
+#start month of the rainy season
+start_rainy_seas=10
+#season approx indicates the year during which the rainy season started
+#this is done because it can start during one year and continue the next calendar year
+#we therefore prefer to group by rainy season instead of by calendar year
+df_for["season_approx"]=np.where(df_for.date.dt.month>=start_rainy_seas,df_for.date.dt.year,df_for.date.dt.year-1)
+df_for
+```
+
+### Adding Region and Months
+
+#### Both January and February
+
+Unsure why the threshold of 210mm cannot be replicated.
+
+
+```python
+#define areas, months, years, and leadtimes of interest for the trigger
+sel_adm=["Southern"]
+sel_months=[1,2]
+seas_years=range(start_year,end_year)
+sel_leadtime=[3]
+
+adm_str="".join([a.lower() for a in sel_adm])
```

Comment:
Large parts from here till line 361 are repeated again for the jan and feb separate computations. So I would create a function where those repetititions are included. As the repetition is quite error-prone (though there doesn't seem to be any errors atm)